src/Interceptor/Impl/CacheInterceptor.php

```diff
@@ -186,18 +188,55 @@ private function buildCacheKey(Instance $instance, Cache $attribute): string
             $identifier
         );
 
-        $identifier .= $this->getMethodCode($method->getReflection());
+        $identifier .= $attribute->ttl;
+        $identifier .= $this->getMethodInnerCode($method->getReflection());
+        $identifier .= $method->getReflection()->getReturnType() !== null
```

Comment:
We should use Symfony Property Info extractor to get the return type to be able to support generics (iterable<Response>, Response[] ...)